core/src/main/java/org/kiwix/kiwixmobile/core/main/ToolbarScrollingKiwixWebView.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import org.kiwix.kiwixmobile.core.utils.DimenUtils.getToolbarHeight
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+
+@SuppressLint("ViewConstructor")
+class ToolbarScrollingKiwixWebView(
+  context: Context,
+  callback: WebViewCallback,
+  attrs: AttributeSet,
+  nonVideoView: ViewGroup,
+  videoView: ViewGroup,
+  webViewClient: CoreWebViewClient,
+  private val toolbarView: View,
+  private val bottomBarView: View,
+  override var sharedPreferenceUtil: SharedPreferenceUtil
+) : KiwixWebView(context, callback, attrs, nonVideoView, videoView, webViewClient) {
+  private val toolbarHeight = getContext().getToolbarHeight()
+  private var parentNavigationBar: View? = null
+  private var startY = 0f
+
+  constructor(
+    context: Context,
+    callback: WebViewCallback,
+    attrs: AttributeSet,
+    nonVideoView: ViewGroup,
+    videoView: ViewGroup,
+    webViewClient: CoreWebViewClient,
+    toolbarView: View,
+    bottomBarView: View,
+    parentNavigationBar: View?,
+    sharedPreferenceUtil: SharedPreferenceUtil
+  ) : this(
+    context, callback, attrs, nonVideoView, videoView, webViewClient, toolbarView,
+    bottomBarView, sharedPreferenceUtil
+  ) {
+    this.parentNavigationBar = parentNavigationBar
```

Comment:
these constructors should be flipped, `parentNavigationBar` should be a val in the primary constructor and the constructor without a `parentNavigationBar` should pass in a null to the primary constructor. Optionally we could also have null as the default argument and let `@JvmOverloads` dupe out the constructors for us

Reply:
@gouri-panda I think this comment was overlooked, do you need me to go into more detail?

Reply:
@macgills Did i do right [here](https://github.com/kiwix/kiwix-android/pull/2465/commits/cdece7a604fe7769b5e9d4717f6600f1933c8396)?